app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -140,6 +99,55 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
     )
   }
 
+  private fun addRecentlyPlayedStoryListSection(
+    recentlyPlayedStoryList: MutableList<PromotedStory>
+  ) {
+    val recentSectionTitleViewModel =
+      SectionTitleViewModel(activity.getString(R.string.ongoing_story_last_week), false)
+    itemList.add(recentSectionTitleViewModel)
+    for (promotedStory in recentlyPlayedStoryList) {
+      val ongoingStoryViewModel = getOngoingStoryViewModel(promotedStory)
+      itemList.add(ongoingStoryViewModel)
+    }
+  }
+
+  private fun getOngoingStoryViewModel(promotedStory: PromotedStory): RecentlyPlayedItemViewModel {
+    return OngoingStoryViewModel(
+      promotedStory,
+      entityType,
+      fragment as OngoingStoryClickListener
+    )
+  }
+  private fun addOlderStoryListSection(olderPlayedStoryList: List<PromotedStory>) {
+    val showDivider = itemList.isNotEmpty()
+    val olderSectionTitleViewModel =
+      SectionTitleViewModel(
+        activity.getString(R.string.ongoing_story_last_month),
+        showDivider
+      )
+    itemList.add(olderSectionTitleViewModel)
+    for (promotedStory in olderPlayedStoryList) {
+      val ongoingStoryViewModel = getOngoingStoryViewModel(promotedStory)
+      itemList.add(ongoingStoryViewModel)
+    }
+  }
+
+  private fun addRecommendedStoryListSection(suggestedStoryList: List<PromotedStory>) {
+    binding.recentlyPlayedToolbar.title = activity.getString(R.string.stories_for_you)
```

Comment:
Suggest updating the toolbar outside this method (e.g. in the subscribe method above) for all the cases to keep it consistent.

Also, don't we need to set this for each of the different cases?

Reply:
We need this for `addRecentlyPlayedStoryListSection ` function too.

Reply:
Done for all the cases.